app/models/notification_preferences.rb

```diff
@@ -74,4 +90,22 @@ def set_defaults
   end
 
   delegate :preferences_to_hash, to: :class
+
+  def transform_boolean(value)
+    case value
+    when "false"
+      false
+    when "true"
+      true
+    else
+      value
+    end
+  end
```

Comment:
I'm surprised we need this. Don't we (or Rails) have some util class to handle boolean inputs?

Reply:
Yeah, I was also surprised...

In other occasions I think we use `ActiveModel::Type::Boolean.new.cast` but it acts a bit differently, and (to my liking) a bit confusing.